railties/lib/rails/tasks/yarn.rake

```diff
@@ -10,17 +10,21 @@ namespace :yarn do
     end
 
     yarn_flags =
-      if `#{Rails.root}/bin/yarn --version`.start_with?("1")
+      if `"#{Rails.root}/bin/yarn" --version`.start_with?("1")
         "--no-progress --frozen-lockfile"
       else
         "--immutable"
       end
 
-    system({ "NODE_ENV" => node_env }, "#{Rails.root}/bin/yarn install #{yarn_flags}")
+    system({ "NODE_ENV" => node_env }, "\"#{Rails.root}/bin/yarn\" install #{yarn_flags}")
+  rescue Errno::ENOENT
+    $stderr.puts "bin/yarn was not found."
+    $stderr.puts "Please run `bundle exec rails app:update:bin` to create it."
+    exit 1
   end
 end
 
 # Run Yarn prior to Sprockets assets precompilation, so dependencies are available for use.
-if Rake::Task.task_defined?("assets:precompile")
+if Rake::Task.task_defined?("assets:precompile") && File.exist?(Rails.root.join("bin", "yarn"))
```

Comment:
The only way of this happen is if `yarn` is not installed and `sprockets` is, but `--skip-javascript` doesn't install or even load sprockets. I think this check here is valid though.

Reply:
I'm confused. Sprockets is still loaded for CSS, right?

Reply:
Right. `--skip-javascript` is not `--skip-assets`. I was mixing both